AK/StdLibExtras.h

```diff
@@ -125,6 +125,17 @@ constexpr bool is_constant_evaluated()
 #endif
 }
 
+template <unsigned long size, unsigned long align>
+struct aligned_storage {
+  struct alignas(align) storage_ {
+    char buffer[size];
+  };
+  using type = storage_;
+};
+
+template <unsigned long size, unsigned long align>
+using aligned_storage_t = aligned_storage<size, align>::type;
```

Comment:
I think the common pattern for `AK` is to put the implementation details in the `AK::Detail` namespace, and then just expose the final type. 

So this would end up looking like:

```cpp
template <unsigned long Size, unsigned long Align>
using AlignedStorage = typename Detail::AlignedStorage<Size, Align>::Type;
```

Reply:
Yup, definitely this. The main declaration can say in `StdLibExtras.h`, but the actual struct definition needs to be moved to `StdLibExtraDetails.h`.

and `s/unsigned long/size_t/g`

Reply:
size_t wasn't available there yet; erred to unsigned long. Will fix.

Reply:
Turns out size_t is defined in Types.h, which includes StdLibExtras.h. There is no way to get size_t in here. Replaced it with __SIZE_TYPE__ which is functionally equivalent...